superset/views/base.py

```diff
@@ -346,9 +347,21 @@ def common_bootstrap_payload() -> Dict[str, Any]:
     messages = get_flashed_messages(with_categories=True)
     locale = str(get_locale())
 
+    # should not expose API TOKEN to frontend
+    frontend_config = {k: conf.get(k) for k in FRONTEND_CONF_KEYS}
+    if conf.get("SLACK_API_TOKEN"):
+        frontend_config["ALERT_REPORTS_NOTIFICATION_METHODS"] = [
+            ReportRecipientType.EMAIL,
+            ReportRecipientType.SLACK,
+        ]
+    else:
+        frontend_config["ALERT_REPORTS_NOTIFICATION_METHODS"] = [
+            ReportRecipientType.EMAIL,
+        ]
```

Comment:
why does this need to live in the frontend config/bootstrap data? i'm not super familiar with the reports/alerts code, but it seems like they might call apis on load, which would mean we could just get this configuration via an api request instead of bundling it in the bootstrap data on every request.

Reply:
this `bootstrap_data` is used by all entry point, includes spa.html. I think populate it with configs not a bad idea.